agent/src/workload_scheduler/dependency_state_validator.rs

```diff
@@ -0,0 +1,239 @@
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use common::objects::{DeletedWorkload, FulfilledBy, WorkloadSpec};
+
+#[cfg_attr(test, mockall_double::double)]
+use crate::parameter_storage::ParameterStorage;
+
+#[cfg(test)]
+use mockall::automock;
+
+pub struct DependencyStateValidator {}
+
+#[cfg_attr(test, automock)]
+impl DependencyStateValidator {
+    pub fn create_fulfilled(workload: &WorkloadSpec, workload_state_db: &ParameterStorage) -> bool {
+        workload
+            .dependencies
+            .iter()
+            .all(|(dependency_name, add_condition)| {
+                workload_state_db
+                    .get_state_of_workload(dependency_name)
+                    .map_or(false, |wl_state| add_condition.fulfilled_by(wl_state))
+            })
+    }
+
+    pub fn delete_fulfilled(
+        workload: &DeletedWorkload,
+        workload_state_db: &ParameterStorage,
+    ) -> bool {
+        workload
+            .dependencies
+            .iter()
+            .all(|(dependency_name, delete_condition)| {
+                workload_state_db
+                    .get_state_of_workload(dependency_name)
+                    .map_or(true, |wl_state| delete_condition.fulfilled_by(wl_state))
+            })
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////
+//                 ########  #######    #########  #########                //
+//                    ##     ##        ##             ##                    //
+//                    ##     #####     #########      ##                    //
+//                    ##     ##                ##     ##                    //
+//                    ##     #######   #########      ##                    //
+//////////////////////////////////////////////////////////////////////////////
+
+#[cfg(test)]
+mod tests {
+    use super::DependencyStateValidator;
+    use common::{
+        objects::{
+            generate_test_workload_spec_with_dependencies, generate_test_workload_spec_with_param,
+            AddCondition, DeleteCondition, ExecutionState,
+        },
+        test_utils::generate_test_deleted_workload_with_dependencies,
+    };
+    use mockall::predicate;
+    use std::collections::HashMap;
+
+    use crate::parameter_storage::MockParameterStorage;
+
+    const AGENT_A: &str = "agent_A";
+    const WORKLOAD_NAME_1: &str = "workload_1";
+    const WORKLOAD_NAME_2: &str = "workload_2";
+    const RUNTIME: &str = "runtime";
+
+    #[test]
+    fn utest_create_fulfilled() {
+        let workload_with_dependencies = generate_test_workload_spec_with_dependencies(
+            AGENT_A,
+            WORKLOAD_NAME_1,
+            RUNTIME,
+            HashMap::from([(WORKLOAD_NAME_2.to_string(), AddCondition::AddCondRunning)]),
+        );
+
+        let execution_state = Box::leak(Box::new(Some(ExecutionState::running())));
```

Comment:
As it is used in the utests only and the size is small leaking memory for comfort is ok in this case.